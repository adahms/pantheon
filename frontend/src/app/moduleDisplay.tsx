import React, { Component } from 'react';
import { Button } from '@patternfly/react-core';
import { Breadcrumb, BreadcrumbItem, BreadcrumbHeading } from '@patternfly/react-core';
import { Link } from "react-router-dom";
import { Level, LevelItem } from '@patternfly/react-core';
import { Tooltip } from '@patternfly/react-core';
import {
    DataList, DataListItem, DataListItemRow, DataListItemCells,
    DataListCell, Card, Text, TextContent, TextVariants, Grid, GridItem
  } from '@patternfly/react-core';
import { Revisions } from '@app/revisions';
import { HelpIcon } from '@patternfly/react-icons';

export interface IProps {
    moduleName: string
    modulePath: string
    moduleType: string
    updated: string    
  }
  
  class ModuleDisplay extends Component<IProps> {

    public state = {
        draftUpdateDate: '',
        releaseUpdateDate: '',
    };

    public render() {        
        return (  
            <React.Fragment>
                <div>
                    <Breadcrumb>
                        <BreadcrumbItem to="#">Modules</BreadcrumbItem>
                        <BreadcrumbItem to="#" isActive={true}>
                            {this.props.moduleName}
                        </BreadcrumbItem>
                    </Breadcrumb>
                </div>
                <div>
                    <Level gutter="md">
                        <LevelItem>
                                <TextContent>
                                    <Text component={TextVariants.h1}>{this.props.moduleName}{'  '}
                                    <Tooltip
                                                position="right"
                                                content={
                                                    <div>Title updated in latest revision</div>
                                                }>
                                            <span><HelpIcon/></span>
                                        </Tooltip>
                                    </Text>  
                                </TextContent>
                        </LevelItem>
                        <LevelItem />
                        <LevelItem>
                            <Button variant="secondary">Edit metadata</Button>
                        </LevelItem>
                    </Level>                
                </div>
                <div>
                    <a href='http://access.redhat.com'>View on Customer Portal</a>
                </div>
                <div>
                    <DataList aria-label="single action data list example ">
                        <DataListItem aria-labelledby="simple-item1">
                            <DataListItemRow id="data-rows-header" >
                                <DataListItemCells
                                    dataListCells={[
                                        <DataListCell width={2} key="products">
                                            <span className="sp-prop-nosort" id="span-source-type">Products</span>                                            
                                        </DataListCell>,
                                        <DataListCell key="published">
                                            <span className="sp-prop-nosort" id="span-source-type">Published</span>
                                        </DataListCell>,
                                        <DataListCell key="updated">
                                            <span className="sp-prop-nosort" id="span-source-type">Draft Uploaded</span>
                                        </DataListCell>,
                                        <DataListCell key="module_type">
                                            <span className="sp-prop-nosort" id="span-source-name">Module Type</span>
                                        </DataListCell>
                                    ]}
                                />
                            </DataListItemRow>

                            <DataListItemRow>
                                <DataListItemCells
                                    dataListCells={[
                                        <DataListCell width={2} key="products">
                                            <span>Dummy Product Name</span>
                                        </DataListCell>,
                                        <DataListCell key="published">
                                            <span>{this.state.releaseUpdateDate.substring(4,15)}</span>
                                        </DataListCell>,
                                        <DataListCell key="updated">
                                            <span>{this.state.draftUpdateDate.substring(4,15)}</span>
                                        </DataListCell>,
                                        <DataListCell key="module_type">
                                            <span>{this.props.moduleType}</span>
                                        </DataListCell>,
                                    ]}
                                />
                            </DataListItemRow>
                            ))}
                    </DataListItem>
                </DataList>
                </div>
                <div>
                    <Card>
                        <Revisions 
                            modulePath={this.props.modulePath}
                            revisionModulePath={this.props.moduleName}
                            draftUpdateDate={this.updateDate}
                            releaseUpdateDate={this.updateDate}
                        />
                    </Card>
                </div> 
            </React.Fragment>

        );
    }

    private updateDate = (date,type) => {
        if(type==="draft"){
            this.setState({
                draftUpdateDate: date
            },() => {
                console.log('changed draft date: ', this.state.draftUpdateDate)
            });    
        }
        else{
            this.setState({
                releaseUpdateDate: date
            },() => {
                console.log('changed release date: ', this.state.releaseUpdateDate)
            });
        }
      };

    
}

export { ModuleDisplay }