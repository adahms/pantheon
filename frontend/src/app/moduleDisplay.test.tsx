import React from 'react';
import { ModuleDisplay } from '@app/moduleDisplay';
import "isomorphic-fetch"

import { mount, shallow } from 'enzyme';
import { Breadcrumb, Button, Card, DataList, DataListItem, DataListItemCells, DataListItemRow, FormGroup, FormSelect, Modal, TextInput } from '@patternfly/react-core';
import renderer from 'react-test-renderer';
import sinon from 'sinon'

const props = {
    moduleName: "Red Hat Enterprise Linux",
    modulePath: "module/red_hat_enterprise_linux",
    moduleType: "REFERENCE",
    updated: "Tue Aug 27 2019 11:28:26 GMT-0400"
}

describe('ModuleDisplay tests', () => {
    test('should render ModuleDisplay component', () => {
        const view = shallow(<ModuleDisplay {...props} />);
        expect(view).toMatchSnapshot();
    });

    it('should render a Breadcrumb', () => {
        const wrapper = mount(<ModuleDisplay {...props} />);
        const breadcrum = wrapper.find(Breadcrumb);
        expect(breadcrum.exists()).toBe(true)
    });

    it('should render a Button', () => {
        const wrapper = mount(<ModuleDisplay {...props} />);
        const button = wrapper.find(Button);
        expect(button.exists()).toBe(true)
    });

    it('should render a Card', () => {
        const wrapper = mount(<ModuleDisplay {...props} />);
        const card = wrapper.find(Card);
        expect(card.exists()).toBe(true)
    });

    it('should render a Data List', () => {
        const wrapper = mount(<ModuleDisplay {...props} />);
        const dataList = wrapper.find(DataList);
        expect(dataList.exists()).toBe(true)
    });

    it('should render a DataListItem', () => {
        const wrapper = mount(<ModuleDisplay {...props} />);
        const dataListItem = wrapper.find(DataListItem);
        expect(dataListItem.exists()).toBe(true)
    });

    it('should render a DataListItemCells Element', () => {
        const wrapper = mount(<ModuleDisplay {...props} />);
        const dataListItemCells = wrapper.find(DataListItemCells);
        expect(dataListItemCells.exists()).toBe(true)
    });

    it('should render a DataListItemRow element', () => {
        const wrapper = mount(<ModuleDisplay {...props} />);
        const dataListItemRow = wrapper.find(DataListItemRow);
        expect(dataListItemRow.exists()).toBe(true)
    });

    // it('should render a form group', () => {
    //     const wrapper = mount(<ModuleDisplay {...props} />);
    //     const formGroup = wrapper.find(FormGroup);
    //     expect(formGroup.exists()).toBe(true)
    // });

    it('should render a Modal', () => {
        const wrapper = mount(<ModuleDisplay {...props} />);
        const modal = wrapper.find(Modal);
        expect(modal.exists()).toBe(true)
    });

    // it('should render a FormSelect Element', () => {
    //     const wrapper = mount(<ModuleDisplay {...props} />);
    //     wrapper.setState({moduleName: "test", modulePath: "modules/test", login: true})
    //     const formSelect = wrapper.find(FormSelect);
    //     expect(formSelect.exists()).toBe(true)
    // });

    // it('should render a text input', () => {
    //     const wrapper = mount(<ModuleDisplay {...props} />);
    //     wrapper.setState({moduleName: "test", modulePath: "modules/test"})
    //     const textInput = wrapper.find(TextInput);
    //     expect(textInput.exists()).toBe(true)
    // });

    // it('test onChangeProduct function', () => {
    //     const wrapper = renderer.create(<ModuleDisplay {...props} />);
    //     wrapper.setState({moduleName: "test", modulePath: "modules/test"})
    //     const inst = wrapper.getInstance();
    //     expect(inst.onChangeProduct("testProduct", event)).toMatchSnapshot();
    // });

    // it('test onChangeVersion function', () => {
    //     const wrapper = renderer.create(<ModuleDisplay {...props} />);
    //     wrapper.setState({moduleName: "test", modulePath: "modules/test"})
    //     const inst = wrapper.getInstance();
    //     expect(inst.onChangeVersion("testVersion", event)).toMatchSnapshot();
    // });

    // it('test onChangeUsecase function', () => {
    //     const wrapper = renderer.create(<ModuleDisplay {...props} />);
    //     wrapper.setState({moduleName: "test", modulePath: "modules/test"})
    //     const inst = wrapper.getInstance();
    //     expect(inst.onChangeUsecase("testUsecase", event)).toMatchSnapshot();
    // });

    // it('test handleURLInput function', () => {
    //     const wrapper = renderer.create(<ModuleDisplay {...props} />);
    //     wrapper.setState({moduleName: "test", modulePath: "modules/test"})
    //     const inst = wrapper.getInstance();
    //     expect(inst.handleURLInput("testURL")).toMatchSnapshot();
    // });

    // it('test productUrlExist function', () => {
    //     const wrapper = renderer.create(<ModuleDisplay {...props} />);
    //     wrapper.setState({moduleName: "test", modulePath: "modules/test"})
    //     const inst = wrapper.getInstance();
    //     expect(inst.productUrlExist("testURL")).toMatchSnapshot();
    // });

    // it('test getProductUrl function', () => {
    //     const wrapper = renderer.create(<ModuleDisplay {...props} />);
    //     wrapper.setState({moduleName: "test", modulePath: "modules/test"})
    //     const inst = wrapper.getInstance();
    //     expect(inst.getProductUrl("testurl")).toMatchSnapshot();
    // });

    // it('test fetchProductVersionDetails function', () => {
    //     const wrapper = renderer.create(<ModuleDisplay {...props} />);
    //     wrapper.setState({moduleName: "test", modulePath: "modules/test"})
    //     const inst = wrapper.getInstance();
    //     expect(inst.fetchProductVersionDetails()).toMatchSnapshot();
    // });
    
    // it('test handleModalToggle function', () => {
    //     const wrapper = renderer.create(<ModuleDisplay {...props} />);
    //     const inst = wrapper.getInstance();
    //     expect(inst.handleModalToggle()).toMatchSnapshot();
    // });

    // it('test loginRedirect function', () => {
    //     const wrapper = renderer.create(<ModuleDisplay {...props} />);
    //     const inst = wrapper.getInstance();
    //     expect(inst.loginRedirect).toMatchSnapshot();
    // });

    // it('test checkAuth function', () => {
    //     const wrapper = renderer.create(<ModuleDisplay {...props} />);
    //     const inst = wrapper.getInstance();
    //     expect(inst.checkAuth).toMatchSnapshot();
    // });

    // it('should handle state changes for login', () => {
    //     const wrapper = shallow(<ModuleDisplay {...props} />)

    //     expect(wrapper.state('login')).toBe(false)
    //     wrapper.setState({ 'login': true })
    //     expect(wrapper.state('login')).toBe(true)
    // });

});