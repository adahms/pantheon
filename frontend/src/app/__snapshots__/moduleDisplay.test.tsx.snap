// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModuleDisplay tests should render ModuleDisplay component 1`] = `
<Fragment>
  <div>
    <Breadcrumb>
      <BreadcrumbItem
        to="#"
      >
        Modules
      </BreadcrumbItem>
      <BreadcrumbItem
        isActive={true}
        to="#"
      >
        Red Hat Enterprise Linux
      </BreadcrumbItem>
    </Breadcrumb>
  </div>
  <div>
    <Level
      gutter="md"
    >
      <LevelItem>
        <TextContent>
          <Text
            component="h1"
          >
            Red Hat Enterprise Linux
              
            <Tooltip
              appendTo={[Function]}
              aria="describedby"
              boundary="window"
              className=""
              content={
                <div>
                  Title updated in latest revision
                </div>
              }
              distance={15}
              enableFlip={true}
              entryDelay={500}
              exitDelay={500}
              flipBehavior={
                Array [
                  "top",
                  "right",
                  "bottom",
                  "left",
                  "top",
                  "right",
                  "bottom",
                ]
              }
              isAppLauncher={false}
              maxWidth="18.75rem"
              position="right"
              trigger="mouseenter focus"
              zIndex={9999}
            >
              <span>
                <HelpIcon
                  color="currentColor"
                  noVerticalAlign={false}
                  size="sm"
                  title={null}
                />
              </span>
            </Tooltip>
          </Text>
        </TextContent>
      </LevelItem>
    </Level>
  </div>
  <div>
    <a
      href="http://access.redhat.com"
    >
      View on Customer Portal
    </a>
  </div>
  <div>
    <DataList
      aria-label="single action data list example "
    >
      <DataListItem
        aria-labelledby="simple-item1"
      >
        <DataListItemRow
          id="data-rows-header"
        >
          <DataListItemCells
            dataListCells={
              Array [
                <DataListCell
                  width={2}
                >
                  <span
                    className="sp-prop-nosort"
                    id="span-source-type"
                  >
                    Products
                  </span>
                </DataListCell>,
                <DataListCell>
                  <span
                    className="sp-prop-nosort"
                    id="span-source-type"
                  >
                    Published
                  </span>
                </DataListCell>,
                <DataListCell>
                  <span
                    className="sp-prop-nosort"
                    id="span-source-type"
                  >
                    Draft Uploaded
                  </span>
                </DataListCell>,
                <DataListCell>
                  <span
                    className="sp-prop-nosort"
                    id="span-source-name"
                  >
                    Module Type
                  </span>
                </DataListCell>,
              ]
            }
          />
        </DataListItemRow>
        <DataListItemRow>
          <DataListItemCells
            dataListCells={
              Array [
                <DataListCell
                  width={2}
                >
                  <span>
                    Dummy Product Name
                  </span>
                </DataListCell>,
                <DataListCell>
                  <span>
                    
                  </span>
                </DataListCell>,
                <DataListCell>
                  <span>
                    
                  </span>
                </DataListCell>,
                <DataListCell>
                  <span>
                    REFERENCE
                  </span>
                </DataListCell>,
              ]
            }
          />
        </DataListItemRow>
        ))}
      </DataListItem>
    </DataList>
  </div>
  <div>
    <Card>
      <Revisions
        draftUpdateDate={[Function]}
        modulePath="module/red_hat_enterprise_linux"
        releaseUpdateDate={[Function]}
        revisionModulePath="Red Hat Enterprise Linux"
      />
    </Card>
  </div>
</Fragment>
`;
